### PR TITLE
feat(extensions): add chrome.tabs.connect API

### PR DESCRIPTION
**Pull request by samuelmaddock**
#### Description of Change
Implements `chrome.tabs.connect` extension API for background pages.

Since Electron extensions already use `//chrome/renderer/extensions/tabs_hooks_delegate.cc` (see #21779), all that was needed for this feature was to copy the function schema into tabs.json.

Electron maintainers, please add the `9-x-y` backport label if you're able to. 🙇 

ref #19447

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by zcbenz** [comment]
Can you amend the commit and force push? There is a bug with lint task that makes it pending forever.

File: spec-main/extensions-spec.ts [hunk 1]
```diff
@@ -147,6 +147,21 @@ ifdescribe(process.electronBinding('features').isExtensionsEnabled())('chrome ex
...
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, nodeIntegration: true } })
+      await w.loadURL(url)
+
+      const portName = require('uuid').v4()
+      const message = { method: 'connectTab', args: [portName] }
+      w.webContents.executeJavaScript(`window.postMessage('${JSON.stringify(message)}', '*')`)
+
+      const [,, responseString] = await emittedOnce(w.webContents, 'console-message')
+
+      expect(responseString).to.equal(portName)
```
Comment (nornagon) [inline spec-main/extensions-spec.ts]: Does this test that the resulting port works (i.e. you can send messages across it)?

Comment (samuelmaddock) [reply spec-main/extensions-spec.ts]: It only tests that the port was created, good catch! I'll send a message through to verify.

**Comment by jkleinsc** [comment]
The Releases WG has approved this for backporting to 9-x-y.

**Pull request merged**